test/FourBitTwoDisclosureDeviceUnlockerTest.java

```diff
@@ -1,24 +1,45 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.List;
 
+import static org.junit.Assert.fail;
 import static org.mockito.Mockito.*;
 
+@SuppressWarnings("ALL")
 public class FourBitTwoDisclosureDeviceUnlockerTest {
     @Test
     public void nullDeviceTest() {
         FourBitTwoDisclosureDeviceUnlocker.unlock(null);
     }
 
     @Test
```

Comment:
This test only tests to ensure that all possible combinations for a 4 bit, 2 disclosed device is correct.
I was considering testing for other types of devices, but this is one implementation.

I thought for a moment it was silly to write an algorithm to produce these combinations, until I realized we need to make it scalable for different devices. For that purpose, we use an algorithm to do it for us.